semgrep-core/src/spacegrep/src/lib/Pre_match.ml

```diff
@@ -0,0 +1,73 @@
+(*
+   Fast check that tells whether a pattern has any chance of matching the
+   document with 'Match.search'.
+*)
+
+(*
+   Check if the pattern contains 'Byte' tokens, since they often don't
+   and the target often does. This is meant as an optimization to not
+   fill the lookup table with many useless entries.
```

Comment:
The comments does not align with what I see in the rest of the code, or maybe I am not getting it correctly? The only place where this is used is in:

> if check_bytes then Hashtbl.remove chars k

So it is not preventing anything from being added too the lookup table.

Reply:
Maybe before you were building the lookup table from the document? Then it would make sense. But the lookup table is built from the pattern.